pyccel/errors/messages.py

```diff
@@ -27,6 +27,7 @@
 INCOMPATIBLE_ARGUMENT = 'Incompatible argument {} passed to function (expected {}). Please cast the argument explicitly or overload the function (see https://github.com/pyccel/pyccel/blob/master/tutorial/headers.md for details)'
 
 UNSUPPORTED_ARRAY_RETURN_VALUE = 'Array return arguments are currently not supported'
+UNSUPPORTED_ARRAY_RANK = 'Arrays of dimensions > 15 are currently not supported'
```

Comment:
I thought your implementation worked up to 16...

Reply:
@saniQQ It appears that you have written also a macro for 16 array dimensions:

```python
#define GET_INDEX_EXP16(t, arr, a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p) GET_INDEX_EXP15(t, arr, a, b, c, d, e, f, g, h, i, j, k, l, m, n, o) + t(arr, 15, p)
```

So why raising an error if the user wants to use 16 indices?

Reply:
I will remove that, i did not notice it.

Reply:
Is it better to stop at 15 or have more dimensions?

Reply:
As per our discussion the other day Fortran2008 says that Fortran is limited to 15 dimensions so this seems like a sensible place to stop